netbox/dcim/models/cables.py

```diff
@@ -518,9 +518,15 @@ def from_origin(cls, terminations):
             # Terminations must all be of the same type
             assert all(isinstance(t, type(terminations[0])) for t in terminations[1:])
 
+            # All mid-span terminations must all be attached to the same device
+            if not isinstance(terminations[0], PathEndpoint):
+                assert all(t.device == terminations[0].device for t in terminations[1:])
```

Comment:
Not all terminations are device components (i.e. CircuitTerminations). This should check that all terminations have the same `parent` rather than `device`.